### PR TITLE
Added support of action to reconnect branch at both side action 

### PR DESCRIPTION
**Pull request by annetill**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No. Only line opening at both sides is supported as remedial action in main branch. An error message throws if a user wants to simulate a one side modification or a line re-connection at both sides. During this feature supports, I rework on the action in `powsybl-core` because it is not adapted to open/close at one side only (https://github.com/powsybl/powsybl-core/pull/2843/).

**What kind of change does this PR introduce?**

For the moment, only the support of a remedial action that connects at both side a line that was totally disconnected.

**What is the current behavior?**

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java [hunk 1]
```diff
@@ -290,6 +290,20 @@ protected static void findAllRtcToOperate(List<Action> actions, LfTopoConfig top
         }
     }
 
+    protected static void findAllBranchToOpen(Network network, List<Action> actions, LfTopoConfig topoConfig) {
```
Comment (rosiereflo) [inline src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java:293]: If I understand well, should be named findAllBranchToClose?

Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java:293]: Ah yes, thanks!

File: src/main/java/com/powsybl/openloadflow/network/impl/Networks.java [hunk 2]
```diff
@@ -124,15 +124,23 @@ private static void retainAndCloseNecessarySwitches(Network network, LfTopoConfi
                 .forEach(sw -> sw.setRetained(true));
 
         topoConfig.getSwitchesToClose().forEach(sw -> sw.setOpen(false)); // in order to be present in the network.
+        topoConfig.getBranchIdsToClose().stream().map(id -> network.getBranch(id)).forEach(branch -> {
```
Comment (obrix) [inline src/main/java/com/powsybl/openloadflow/network/impl/Networks.java:127]: .map(id -> network.getBranch(id)) ---> .map(network::getBranch) (SONAR)

**Review by obrix — COMMENTED** [review]
Looks good to me, just a small fix needed for sonar.

**Review by rosiereflo — APPROVED** [review]

**Review by jeandemanged — APPROVED** [review]
I'd just rename the PR to "Added support of branch re-connection action" if you agree

**Pull request merged**